AutoRaise.mm

```diff
@@ -475,48 +475,140 @@ CGEventRef eventTapHandler(CGEventTapProxy proxy, CGEventType type, CGEventRef e
+            
+            // old style config files
+            if (isFileExist(oldStyleDelayFile) or isFileExist(oldStyleWarpFile)) {
+                NSFileHandle * delayFile = [NSFileHandle fileHandleForReadingAtPath: oldStyleDelayFile];
+                if (delayFile) {
+                    delayCount = abs([[[NSString alloc] initWithData:
+                        [delayFile readDataOfLength: 2] encoding:
+                        NSUTF8StringEncoding] intValue]);
+                    [delayFile closeFile];
+                }
+                NSFileHandle * warpFile = [NSFileHandle fileHandleForReadingAtPath: oldStyleWarpFile];
+                if (warpFile) {
+                    warpMouse = true;
+                    NSString * line = [[NSString alloc] initWithData:
+                        [warpFile readDataOfLength:11] encoding:
+                        NSUTF8StringEncoding];
+                    NSArray * components = [line componentsSeparatedByString: @" "];
+                    if (components.count >= 1) { warpX = [[components objectAtIndex:0] floatValue]; }
+                    if (components.count >= 2) { warpY = [[components objectAtIndex:1] floatValue]; }
+                    if (components.count >= 3) { cursorScale = [[components objectAtIndex:2] floatValue]; }
+                    [warpFile closeFile];
+                }
+                
+            } else {
+                
+            //new style config files (if there are no old style config files)
+                NSString * newStyleConfigFile;
+                if (isFileExist(newStyleConfigFile1)) { newStyleConfigFile = newStyleConfigFile1; }
+                else if (isFileExist(newStyleConfigFile2)) { newStyleConfigFile = newStyleConfigFile2; }
+                
+                if (newStyleConfigFile) {
+                    NSError  * error;
+                    NSString * newStyleConfig = [[NSString alloc]
+                                                initWithContentsOfFile: newStyleConfigFile
+                                                encoding: NSUTF8StringEncoding
+                                                error: &error];
+                    
+                    while ([newStyleConfig containsString:@"  " ]) {
+                        newStyleConfig = [newStyleConfig stringByReplacingOccurrencesOfString:@"  " withString:@" "];
+                    }
+               
+                    NSArray  * newStyleConfigLines = [newStyleConfig componentsSeparatedByString:@"\n"];
+
+                    NSString * newStyleConfigLine;
+                    NSArray  * temporaryConfigDictionary = @[@"delay", @"warpx", @"warpy", @"scale"];
+                    int newStyleConfigDelay = -1;
+                    float newStyleConfigWarpX = -1.0;
+                    float newStyleConfigWarpY = -1.0;
+                    float newStyleConfigScale = -1.0;
+
+                    for ( int i = 0; i < [newStyleConfigLines count]; i++) {
```

Comment:
ah, reading the new format here, line by line. probably worth of its own method as well ;-)

Reply:
Ok. I need to learn something about 'methods' first :-)